Sources/Jobs/Middleware/MetricsMiddleware.swift

```diff
@@ -68,24 +69,42 @@ public struct MetricsJobMiddleware: JobMiddleware {
     ///   - jobInstanceID: Job instance identifer
     @inlinable
     public func onPopJob(result: Result<any JobInstanceProtocol, JobQueueError>, jobInstanceID: String) async {
-        // Decrement the current queue by 1
-        Meter(
-            label: Self.meterLabel,
-            dimensions: [
-                ("status", JobStatus.queued.rawValue)
-            ]
-        ).decrement()
 
         switch result {
         case .failure(let error):
+            var counterDimensions: [(String, String)] = [
+                ("reason", error.code.description),
+                ("jobID", jobInstanceID),
```

Comment:
Sorry one more thing. Can we remove jobID? We shouldn't have a new instance of a dimension for every instance of a job